src/test/java/de/retest/recheck/ui/descriptors/RootElementFilterTest.java

```diff
@@ -0,0 +1,314 @@
+package de.retest.recheck.ui.descriptors;
+
+import static de.retest.recheck.ui.Path.fromString;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import de.retest.recheck.ignore.CompoundFilter;
+import de.retest.recheck.ignore.Filter;
+import de.retest.recheck.review.ignore.AttributeFilter;
+import de.retest.recheck.ui.Path;
+import de.retest.recheck.ui.image.Screenshot;
+import de.retest.recheck.ui.image.Screenshot.ImageType;
+
+class RootElementFilterTest {
+
+	RootElementFilter rootElementFilter;
+
+	@Nested
+	class AttributeFilterTests {
+		private final Element element = mock( Element.class );
+		Filter filter;
+		Attributes state;
+
+		@BeforeEach
+		void setUp() {
+			final MutableAttributes mutable = new MutableAttributes();
+			mutable.put( "enabled", "true" );
+			mutable.put( "rowCount", "5" );
+			state = mutable.immutable();
+		}
+
+		@Test
+		public void should_return_same_attributes_with_filter_nothing() {
+			filter = Filter.FILTER_NOTHING;
+			rootElementFilter = new RootElementFilter( filter );
+			final Attributes newState = rootElementFilter.filter( element, state );
+
+			assertThat( (Object) newState ).isEqualTo( state );
+		}
+
+		@Test
+		public void should_return_same_attributes_with_no_filter() {
+			filter = new CompoundFilter();
+			rootElementFilter = new RootElementFilter( filter );
+			final Attributes newState = rootElementFilter.filter( element, state );
+
+			assertThat( (Object) newState ).isEqualTo( state );
+		}
+
+		@Test
+		public void should_filter_respective_attribute() {
+			filter = new CompoundFilter( new AttributeFilter( Attributes.ENABLED ) );
+			rootElementFilter = new RootElementFilter( filter );
+			final Attributes newState = rootElementFilter.filter( element, state );
+
+			assertThat( newState.size() ).isEqualTo( 1 );
+			assertThat( newState.get( "enabled" ) ).isNull();
+			assertThat( newState.get( "rowCount" ) ).isEqualTo( "5" );
+		}
+
+		@Test
+		public void should_not_do_anything_with_no_matches() {
+			filter = new CompoundFilter( new AttributeFilter( Attributes.FONT_FAMILY ) );
+			rootElementFilter = new RootElementFilter( filter );
+			final Attributes newState = rootElementFilter.filter( element, state );
+
+			assertThat( (Object) newState ).isEqualTo( state );
+		}
+
+	}
+
+	@Nested
+	class IdentifyingAttributesFilterTests {
+		private final Path path = Path.fromString( "Window[1]/path[1]/component[1]" );
```

Comment:
Here `Window[1]/path[1]/component[1]`, below `Window[1]/Parent[1]/Component[2]`. Try to be more consistent.